moped-editor/src/views/projects/projectView/ProjectSummary.js

```diff
@@ -28,7 +106,81 @@ const ProjectSummary = ({ loading, error, data, refetch }) => {
     projectFeatureRecords
   );
 
-  refetch();
+  /**
+   * Handles updating the state for "status update"
+   * @param {Object} event
+   */
+  const handleStatusUpdateChange = event => setStatusUpdate(event.target.value);
+
+  /**
+   * Handles the edit click for status update
+   */
+  const handleStatusUpdateEdit = () => {
+    setStatusUpdateEditable(true);
+  };
+
+  /**
+   * Handles adding a new status update
+   */
+  const handleStatusUpdateAddNew = () => {
+    setStatusUpdateAddNew(true);
+    setStatusUpdateEditable(true);
+    setStatusUpdate("");
+  };
+
+  /**
+   * Handles saving the status update
+   */
+  const handleStatusUpdateSave = () => {
+    // Retrieve a commentId or get a null
+    const commentId = getStatusUpdate("project_note_id");
+    const addedBy = `${userSessionData.first_name} ${userSessionData.last_name}`;
+    const isStatusUpdateInsert = statusUpdateAddNew || !commentId;
+
+    (isStatusUpdateInsert
+      ? updateProjectStatusUpdateInsert
+      : updateProjectStatusUpdateUpdate)({
+      variables: {
+        ...(isStatusUpdateInsert
+          ? {
+              statusUpdate: {
+                project_id: Number(projectId),
+                added_by: addedBy,
+                project_note: statusUpdate,
+                status_id: 1,
+                project_note_type: 2,
+              },
+            }
+          : {
+              project_note_id: { _eq: Number(commentId) },
+              added_by: addedBy,
+              project_note: statusUpdate,
+            }),
+      },
+    })
```

Comment:
This is the core of it, basically it checks whether this is an insert of a new comment, or an update of an existing one, and based on that condition it selects the function to execute and what variables to pass.

Kind of a convoluted summarized function, but I did not want it to be over-repetitive.